src/lib.rs

```diff
@@ -14,6 +14,7 @@ use diagnostics::{Diagnostics, DiagnosticsConfig, ERR_INDENT_SIZE};
 use net::hvm_to_net::hvm_to_net;
 
 pub mod diagnostics;
+#[allow(clippy::mutable_key_type)]
```

Comment:
Add a comment explaining why this is necessary.
Eg. "`Name` is implemented with inner mutability so it triggers this warning, but we know it works."